### PR TITLE
Handle unsupported content types

### PR DESCRIPTION
**Pull request by fabriguespe**
This PR explains how to recognize un supported content types and display a fallback instead.

- [Preview](https://junk-range-possible-git-fallbackv1-xmtp-labs.vercel.app/docs/build/messages/#handle-unsupported-content-types)

### CODE REVIEW

File: docs/build/messages/messages.md [hunk 1]
```diff
@@ -345,70 +345,87 @@ for await (const page of conversation.messages(limit: 25)) {
...
-    message.content === undefined &&
-    ContentTypeId.fromString(message.contentType).sameAs(ContentTypeAttachment)
-  ) {
-    return "This message contains an attachment, which is not supported by this client.";
-  }
-};
+const codec = client.codecFor(content.contentType);
+if (!codec) {
+  /*Not supported content type*/
+  return message?.contentFallback;
```
Comment (nplasterer) [inline docs/build/messages/messages.md:372]: Would it be worth showing how to handled `nil` contentFallback? Or is that handled in a different tutorial?

File: docs/build/messages/messages.md [hunk 2]
```diff
@@ -345,70 +345,87 @@ for await (const page of conversation.messages(limit: 25)) {
...
+  /*Not supported content type*/
+  return message?.contentFallback
+}
+```
 
 </TabItem>
 <TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
 
-Code sample coming soon
+```jsx
```
Comment (nplasterer) [inline docs/build/messages/messages.md:390]:
```suggestion
```swift
```

File: docs/build/messages/messages.md [hunk 3]
```diff
@@ -345,70 +345,87 @@ for await (const page of conversation.messages(limit: 25)) {
...
+}
+```
 
 </TabItem>
 <TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>
 
-Code sample coming soon
+Dart doesnt have a public method for checking if the content type is registered. Instead the decode function uses this method internally and returns the `encoded.fallback` if exists.
+
+```jsx
```
Comment (nplasterer) [inline docs/build/messages/messages.md:403]:
```suggestion
```dart
```

File: docs/build/messages/messages.md (hunk 1 above)
Comment (fabriguespe) [reply docs/build/messages/messages.md:372]: Thanks for the suggestion! Something like this? I can then apply it to the rest of the snippets.
```
const codec = client.codecFor(content.contentType);
if (!codec) {
  /*Not supported content type*/
  if (message?.contentFallback !== undefined) {
    return message?.contentFallback;
  }
  // Handle other types like ReadReceipts which are not mean to be displayed
}
```

File: docs/build/messages/messages.md [hunk 4]
```diff
@@ -345,70 +345,105 @@ for await (const page of conversation.messages(limit: 25)) {
 </TabItem>
 </Tabs>
 
-## Handle an unsupported content type error
+## Handle unsupported content types
 
-As more [custom](/docs/concepts/content-types#create-a-custom-content-type) and [standards-track](/docs/concepts/content-types#standards-track-content-types) content types enter the XMTP ecosystem, your app might receive a content type your app doesn't support. This could crash your app.
+As more [custom](/docs/concepts/content-types#create-a-custom-content-type) and [standards-track](/docs/concepts/content-types#standards-track-content-types) content types are introduced into the XMTP ecosystem, your app may encounter content types it does not support. This situation, if not handled properly, could lead to app crashes.
+
+Each content type includes a `contentFallback` property. This property provides a string that describes the expected value of the content type. Note that content fallbacks are immutable and are set by default in the protocol. If you are creating custom content types, you have the option to include a custom fallback. For more information on this, please visit the [Custom Content Type Tutorial](/docs/tutorials/custom-ct).
```
Comment (rygine) [inline docs/build/messages/messages.md:352]: >  and are set by default in the protocol

this seems a bit confusing. perhaps `and are set by default in network messages`. or maybe, `and are embedded into each message` or similar.

WDYT?

File: docs/build/messages/messages.md [hunk 5]
```diff
@@ -345,70 +345,105 @@ for await (const page of conversation.messages(limit: 25)) {
...
 ```jsx
-const codec = xmtp.codecFor(content.contentType);
+const codec = client.codecFor(content.contentType);
 if (!codec) {
-  const fallback = `missing codec for content type "${content.contentType.toString()}"`;
-  throw new Error(fallback);
+  /*Not supported content type*/
+  if (message?.contentFallback !== undefined) {
+    return message?.contentFallback;
+  }
```
Comment (rygine) [inline docs/build/messages/messages.md:363]: are these optional chains on `message` necessary?

File: docs/build/messages/messages.md [hunk 6]
```diff
@@ -345,70 +345,105 @@ for await (const page of conversation.messages(limit: 25)) {
...
+}
+```
 
 </TabItem>
 <TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
 
-Code sample coming soon
+```swift
+let codec = client.codecRegistry.find(for: contentType)
+if(!codec){
```
Comment (rygine) [inline docs/build/messages/messages.md:401]: inconsistent spacing between `if` and `(` here

File: docs/build/messages/messages.md [hunk 7]
```diff
@@ -345,70 +345,105 @@ for await (const page of conversation.messages(limit: 25)) {
...
+import 'package:xmtp/src/content/codec_registry.dart';
+import 'package:xmtp/src/content/text_codec.dart';
+
+// Create a CodecRegistry and register a TextCodec
+var registry = CodecRegistry();
+registry.registerCodec(TextCodec());
+// Use the registry to decode the content
+var decoded = await registry.decode(encoded);
+//if the content type doesn't exists. will return `encoded.fallback`
+if(decoded == null){
```
Comment (rygine) [inline docs/build/messages/messages.md:426]: inconsistent spacing between `if` and `(` here

File: docs/build/messages/messages.md (hunk 4 above)
Comment (rygine) [inline docs/build/messages/messages.md:352]: should this be:

> Each _message_ includes a `contentFallback` property.

should we also mention that the fallback can also be `undefined`, and how to handle that case?

File: docs/build/messages/messages.md [hunk 8]
```diff
@@ -345,70 +345,105 @@ for await (const page of conversation.messages(limit: 25)) {
...
+As more [custom](/docs/concepts/content-types#create-a-custom-content-type) and [standards-track](/docs/concepts/content-types#standards-track-content-types) content types are introduced into the XMTP ecosystem, your app may encounter content types it does not support. This situation, if not handled properly, could lead to app crashes.
+
+Each content type is accompanied by a `contentFallback` property, which offers a descriptive string representing the content type's expected value. It's important to note that content fallbacks are immutable and are predefined in network messages. In instances where `contentFallback` is `undefined`, it indicates that the content, such as read receipts, is not intended to be rendered. If you're venturing into creating custom content types, you're provided with the flexibility to specify a custom fallback. For a deeper dive into this, consider exploring the [Custom Content Type Tutorial](/docs/tutorials/custom-ct).
```
Comment (rygine) [inline docs/build/messages/messages.md:352]: > Each content type is accompanied by a `contentFallback` property

should this say each message? or is this referring to the codecs, which have a `fallback` property?

**Pull request merged**